### DISCUSSION TITLE
Implementation Fooocus in cloud

### DISCUSSION THREAD

**Discussion by eduardogiorgio** [lllyasviel/Fooocus#2505] [Q&A] [answered]
Hi!
I would like to implement foocus as like sass (Software as a service). 
Preference in Azure.
Could you give me some recomendations?
Thanks

**Comment by mashb1t** [comment]
Please find further information here:
https://github.com/lllyasviel/Fooocus/discussions/2292
https://github.com/lllyasviel/Fooocus/discussions/2302
https://github.com/lllyasviel/Fooocus/discussions/1981
https://github.com/lllyasviel/Fooocus/discussions/1957

**Reply by mashb1t** [reply]
i'd propose to remove the Gradio queue, use a AWS SQS equivalent with listeners and message queue pooling (calling Fooocus as API) as well as a persistent storage layer. Websockets may be forwarded using a message broker and direct linking to the instances. Keep in mind that you need NAT as well as STUN/TURN for mobile users.

**Reply by AndrewMTA5** [reply]
Is there any code samples of it? Could you deploy it on hugging face and use the gradio API?

**Reply by mashb1t** [reply, marked as answer]
@AndrewMTA5 I won't pay for a server for you, feel free to host it yourself.
Be aware that the API will be unprotected and anyone can use it if not configured correctly.

To quickly test the API i'd recommend to use the Colab template in the readme, direct link: https://colab.research.google.com/github/lllyasviel/Fooocus/blob/main/fooocus_colab.ipynb

**Reply by AndrewMTA5** [reply]
@mashb1t  When I run "gradio deploy" it will push to hugging face spaces then it gives me an inaccessible local url, if I set share=True in the launch() function I'll get an error from hugging face saying it's not allowed. 

When I use the Collab it will say "this link share link expires in 72 hours, for permanent hosting use gradio deploy".

I'd like to set up a permanent API to use but it looks like spaces won't let you --share

**Reply by eduardogiorgio** [reply]
Thanks for all. Colab could be work for me.
